src/koza/model/config/source_config.py

```diff
@@ -147,37 +132,166 @@ class DatasetDescription:
+            entity["original_object"] = entity["object"]
+            entity["object"] = self._get_mapping(entity["object"], self.object_target_prefixes)
+
+        return entity
+    
+    def _merge_and_filter_sssom(self):
+        mapping_sets = []
+        for file in self.files:
+            msdf = parse_sssom_table(file)
+            mapping_sets.append(msdf)
+        new_msdf = merge_msdf(*mapping_sets)
+        filters = (self.subject_target_prefixes + self.object_target_prefixes) + list(
+            set(self.filter_prefixes)
+            - set(self.subject_target_prefixes)
+            - set(self.object_target_prefixes)
+        )
+        logger.debug(f"Filtering SSSOM by {filters}")
+        new_msdf = filter_prefixes(
+            df=new_msdf.df,
+            filter_prefixes=filters,
+            require_all_prefixes=False
+        )
+
+        return new_msdf
+
+    def _build_sssom_lut(self) -> Dict:
+        """Build a lookup table from SSSOM mapping dataframe."""
+        sssom_lut = {}
+        for _, row in self.df.iterrows():
+            subject_id = row["subject_id"]
+            object_id = row["object_id"]
+            predicate = row["predicate_id"]
+            sssom_lut = self._set_mapping(subject_id, object_id, predicate, match='broad', lut=sssom_lut)
+            sssom_lut = self._set_mapping(object_id, subject_id, predicate, match='narrow', lut=sssom_lut)
+        return sssom_lut
+
+    def _has_match(self, predicate, match: Literal['narrow', 'broad']):
+        if match == 'narrow':
+            return predicate in self.predicates['narrow'] or predicate in self.predicates['exact']
+        if match == 'broad':
+            return predicate in self.predicates['broad'] or predicate in self.predicates['exact']
+
+    def _has_mapping(self, id, target_prefixes = None):
+        """Check if an ID has a mapping."""
+        if target_prefixes is None:
+            return id in self.lut
+        else:
+            if id not in self.lut:
+                return False
+            for target_prefix in target_prefixes:
+                if target_prefix in self.lut[id]:
+                    return True
+            # logger.warning(f"Could not find mapping in SSSOM lookup table for {id} with any of {target_prefixes}")
+            return False
+    
+    def _get_mapping(self, id, target_prefixes):
+        """Get the mapping for an ID."""
+        for target_prefix in target_prefixes:
+            if target_prefix in self.lut[id]:
+                return self.lut[id][target_prefix]
+        raise KeyError(f"Could not find mapping for {id} in {target_prefixes}: {self.lut[id]}")
+    
+    def _set_mapping(self, original_id, mapped_id, predicate, match: Literal['narrow', 'broad'], lut: Dict[str, dict]):
+        """Set a mapping for an ID."""
+        original_prefix = original_id.split(":")[0]
+        mapped_prefix = mapped_id.split(":")[0]
+        filter_prefixes = self.filter_prefixes
```

Comment:
why is this assigned to a local variable?

Reply:
no reason to be honest. same as the mixed check in your other suggestion, i think it just slipped through the crack and was mostly there for my clarity as i was writing it up. 
removed local in favor of class variable